### PR TITLE
doc: How to create new release branch CI files

### PR DESCRIPTION
**Pull request by mateusoliveira43**
## Why the changes were made

The goal of this PR is to document the steps needed to create a new release branches, so no steps are missed next time we do it. In the future, this should be automated.

Fixes https://github.com/openshift/oadp-operator/issues/1227

## How to test the changes made

Read the [documentation](https://github.com/mateusoliveira43/oadp-operator/blob/doc/ci-new-branch/docs/developer/PROW_CI.md#creating-new-release-branches) and make sure it is clear. Check referenced PRs in linked issue to see if no steps were missed.

### CODE REVIEW

File: docs/developer/PROW_CI.md
Comment (mateusoliveira43) [inline docs/developer/PROW_CI.md:64]: TODO

File: docs/developer/PROW_CI.md
Comment (mateusoliveira43) [inline docs/developer/PROW_CI.md:141]: also update codecov branch

**Comment by mateusoliveira43** [comment]
/remove-lifecycle stale 
documentation is important

**Comment by mateusoliveira43** [comment]
/hold
not ready, want to test CI

File: docs/developer/PROW_CI.md
Comment (mateusoliveira43) [inline docs/developer/PROW_CI.md:197]: wait https://github.com/openshift/oadp-operator/pull/1463 to finish this TODO

**Review by kaovilai — APPROVED** [review]

File: hack/ci.Dockerfile
Comment (shubham-pampattiwar) [inline hack/ci.Dockerfile:1]: Duplicate file ?

Comment (mateusoliveira43) [reply hack/ci.Dockerfile:1]: uhum

after merge I will open a CI PR (in openshift/release) to CI to use this new dockerfile; and then, another PR to remove `build/` folder from here, to fix https://github.com/openshift/oadp-operator/issues/1259

**Review by sseago — APPROVED** [review]

**Comment by mateusoliveira43** [comment]
/override ci/prow/4.15-e2e-test-aws
restore succeeds, but deployment/deploymentConfig never goes to ready state flake

**Comment by mateusoliveira43** [comment]
/override ci/prow/4.16-e2e-test-aws
restore succeeds, but deployment/deploymentConfig never goes to ready state flake

File: docs/developer/PROW_CI.md
Comment (mpryc) [inline docs/developer/PROW_CI.md:70]: What CI Files? Is there a link or information what the`CI files` are ?

File: docs/developer/PROW_CI.md
Comment (mpryc) [inline docs/developer/PROW_CI.md:71]: Do we have a clear list of those repositories? Maybe a link so no repo is forgotten?

File: docs/developer/PROW_CI.md
Comment (mpryc) [inline docs/developer/PROW_CI.md:73]: - Downstream OADP operator always requires updates.
 - Other downstream repositories and their release branches generally do not need updates.
   `- Write when they may require updates`

File: docs/developer/PROW_CI.md
Comment (mpryc) [inline docs/developer/PROW_CI.md:76]: Could we have a little help here:

The default branch of a given repository can be checked using the following command
```sh
$ curl -s https://api.github.com/repos/openshift/oadp-operator | jq -r .default_branch
master
```

File: docs/developer/PROW_CI.md
Comment (mpryc) [inline docs/developer/PROW_CI.md:78]: `can` or `should` be deleted? Leaving option to leave it would be softer then explicitly stating that we only want the default branch to be updated with docs. Does the default branch carries over old design documentation, or proper instructions for a given branch? What if the instructions are changing in master branch? Do we keep the correct ones for a given branch?

File: docs/developer/PROW_CI.md
Comment (mpryc) [inline docs/developer/PROW_CI.md:102]: What about that part of the file? I thought we want to make sure it's inline with the ocp version that the ci is running on?

```
build_root:
  image_stream_tag:
    name: builder
    namespace: konveyor
    tag: ubi9-v1.22.2
```

File: docs/developer/PROW_CI.md
Comment (mpryc) [inline docs/developer/PROW_CI.md:121]: There is no such file in the release folder:
```shell
$ ls ci-operator/config/openshift/oadp-operator/*VERSION*
ls: cannot access 'ci-operator/config/openshift/oadp-operator/*VERSION*': No such file or directory
```

Comment (mpryc) [reply docs/developer/PROW_CI.md:121]: Here is list of files

```shell
$ ls ci-operator/config/openshift/oadp-operator/*
ci-operator/config/openshift/oadp-operator/openshift-oadp-operator-master__4.14.yaml    ci-operator/config/openshift/oadp-operator/openshift-oadp-operator-oadp-1.0.yaml        ci-operator/config/openshift/oadp-operator/openshift-oadp-operator-oadp-1.4__4.14.yaml
ci-operator/config/openshift/oadp-operator/openshift-oadp-operator-master__4.15.yaml    ci-operator/config/openshift/oadp-operator/openshift-oadp-operator-oadp-1.3__4.12.yaml  ci-operator/config/openshift/oadp-operator/openshift-oadp-operator-oadp-1.4__4.15.yaml
ci-operator/config/openshift/oadp-operator/openshift-oadp-operator-master__4.16.yaml    ci-operator/config/openshift/oadp-operator/openshift-oadp-operator-oadp-1.3__4.13.yaml  ci-operator/config/openshift/oadp-operator/openshift-oadp-operator-oadp-1.4__4.16.yaml
ci-operator/config/openshift/oadp-operator/openshift-oadp-operator-master.yaml          ci-operator/config/openshift/oadp-operator/openshift-oadp-operator-oadp-1.3__4.14.yaml  ci-operator/config/openshift/oadp-operator/openshift-oadp-operator-oadp-1.4.yaml
ci-operator/config/openshift/oadp-operator/openshift-oadp-operator-oadp-1.0__4.12.yaml  ci-operator/config/openshift/oadp-operator/openshift-oadp-operator-oadp-1.3__4.15.yaml  ci-operator/config/openshift/oadp-operator/OWNERS
ci-operator/config/openshift/oadp-operator/openshift-oadp-operator-oadp-1.0__4.14.yaml  ci-operator/config/openshift/oadp-operator/openshift-oadp-operator-oadp-1.3.yaml

```

File: docs/developer/PROW_CI.md
Comment (mpryc) [inline docs/developer/PROW_CI.md:221]: Do we help here the developer to find out what go versions are available for a particular OCP or OADP? It's easy to follow the documentation, but it's harder to define which version is correct.

File: docs/developer/PROW_CI.md
Comment (mpryc) [inline docs/developer/PROW_CI.md:376]: This won't always work, one needs to find out which release stream exists. E.g. this 4.16 was available only within:

```shell
releases:
  latest:
    candidate:
      product: ocp
      stream: nightly
      version: "4.16"
```

In the past I've been using the following part to find out if the correct release is within correct stream + grepping other projects and looking which stream does the ocp belongs to:

https://amd64.ocp.releases.ci.openshift.org/api/v1/releasestream/4.16.0-0.nightly/latest

https://amd64.ocp.releases.ci.openshift.org/

File: docs/developer/PROW_CI.md
Comment (mateusoliveira43) [reply docs/developer/PROW_CI.md:70]: yes, in this same file, there are links to each OADP repo CI files

example https://github.com/openshift/oadp-operator/blob/master/docs/developer/PROW_CI.md?plain=1#L5-L8

File: docs/developer/PROW_CI.md
Comment (mateusoliveira43) [reply docs/developer/PROW_CI.md:71]: yes, in this same file, at the top

File: docs/developer/PROW_CI.md
Comment (mateusoliveira43) [reply docs/developer/PROW_CI.md:73]: I do not know if there is a clear when

File: docs/developer/PROW_CI.md
Comment (mateusoliveira43) [reply docs/developer/PROW_CI.md:76]: this can be checked through UI, do not think we need this

File: docs/developer/PROW_CI.md
Comment (mateusoliveira43) [reply docs/developer/PROW_CI.md:78]: we currently do not delete, I want to start doing this

we have old docs in master yes, like this https://github.com/openshift/oadp-operator/blob/master/docs/upgrade_1-1_to_1-2.md

File: docs/developer/PROW_CI.md
Comment (mateusoliveira43) [reply docs/developer/PROW_CI.md:102]: I think this is responsibility of this issue https://github.com/openshift/oadp-operator/issues/1455

File: docs/developer/PROW_CI.md
Comment (mateusoliveira43) [reply docs/developer/PROW_CI.md:121]: VERSION is the minor version of OCP, the regex should be `ci-operator/config/openshift/oadp-operator/openshift-oadp-operator-<branch-name>__4.*.yaml`

File: docs/developer/PROW_CI.md
Comment (mateusoliveira43) [reply docs/developer/PROW_CI.md:221]: developer needs to check `go.mod` of previous version

File: docs/developer/PROW_CI.md
Comment (mateusoliveira43) [reply docs/developer/PROW_CI.md:376]: added note, will be in next commit

File: docs/developer/PROW_CI.md
Comment (mpryc) [reply docs/developer/PROW_CI.md:121]: Ah ok so it's not clear then, it needs to be:

```shell
ci-operator/config/openshift/oadp-operator/openshift-oadp-operator-oadp-1.4__4.${MINOR_VERSION}.yaml
```

File: docs/developer/PROW_CI.md
Comment (mpryc) [reply docs/developer/PROW_CI.md:102]: On the above there is commend `As it is not valid anymore after...`, so either resurrect there or fix here.

File: docs/developer/PROW_CI.md
Comment (mpryc) [reply docs/developer/PROW_CI.md:78]: So I would make this document clear that we do delete and then continue with implementation (deletion). Note, this won't affect rebases and cherry-picks?

File: docs/developer/PROW_CI.md
Comment (mpryc) [reply docs/developer/PROW_CI.md:76]: It can be checked in many way of course. But with new branch you usually use shell or UI? either one works if it's the one to continue. I believe we don't want the user to switch between UI and shell, rather have set of instructions within one.

File: docs/developer/PROW_CI.md
Comment (mpryc) [reply docs/developer/PROW_CI.md:71]: Since it's *.md make it a link?

**Review by kaovilai — COMMENTED** [review]
Can you add info on downstream owners file aliases for repos too?

https://github.com/openshift/release/blob/dd6b8b25a85bfd92ca74fdf1435ee9f21cd22516/core-services/prow/02_config/_plugins.yaml#L664-L678

**Comment by mateusoliveira43** [comment]
@kaovilai where do you suggest adding the info?

**Comment by kaovilai** [comment]
I thought the same readme here. docs/developer/PROW_CI.md

It allows people to issue /override of tests on forks that we do not want to carry changes to OWNERS.md

**Comment by mateusoliveira43** [comment]
@kaovilai done

**Review by kaovilai — APPROVED** [review]

**Review by shubham-pampattiwar — APPROVED** [review]

**Pull request merged**
